### DISCUSSION TITLE
Add support for L band Inmarsat data?

### DISCUSSION THREAD

**Discussion by jprochazka** [jprochazka/adsb-receiver#597] [Ideas]
> This idea is mainly for reference while I dabble with the idea of adding such a feature to the project. At this time there is no guarantee this feature will be added. It comes down to how successful I am in completing the task without doing so interfering too much with more pressing matters. That along with if anyone besides myself is even interested in such data.

While waiting on components needed to complete my over-the-air ACARS j-pole for which I thought I had the remaining needed parts on hand I began toying with the idea of adding the ability to consume L band Inmarsat data using SatDump. I am looking into the possibility of using the SatDump CLI live processing option along with one of the Inmarsat pipelines they supply to generate data which can then be made available via the file system or better yet maybe over UDP? If I understand things correctly this data will be supplied in JSON format which can be easily consumed by either the Python data collection script and/or directly by the portal.

I have put together a proof-of-concept setup using a laptop and some really old Othernet hardware which I bought some time ago which in the past was in fact able to receive Inmarsat data using JAREO on Windows. I am having trouble getting reception from I-4 F3 as well as I-4 F4. It might be a bad LNA because swapping out the OEM unlabeled NooElec RTL2832U RTL-SDR with an old AirSpy Mini did not help things either. I think I will also try to locate a better satellite finding app and maybe change positions to see if that helps but I am thinking I need to start investing in some more up to date hardware in the very near future...

**Comment by MCancellier** [comment]
Joe,Are you in contact with the Airframes.io folks? That’s the best place to send ACARS, VDL2 and the SATCOM modes. You also might want to take a look at @thebaldgeek on GitHub. He has built a network of data feeders. Glad the project is going again!Mike CancellierOn Aug 4, 2024, at 16:14, Joseph Prochazka ***@***.***> wrote:﻿

This idea is mainly for reference while I dabble with the idea of adding such a feature to the project. At this time there is no guarantee this feature will be added. It comes down to how successful I am in completing the task without doing so interfering too much with more pressing matters. That along with if anyone besides myself is even interested in such data.

While waiting on components needed to complete my over-the-air ACARS j-pole for which I thought I had the remaining needed parts on hand I began toying with the idea of adding the ability to consume L band Inmarsat data using SatDump. I am looking into the possibility of using the SatDump CLI live processing option along with one of the Inmarsat pipelines they supply to generate data which can then be made available via the file system or better yet maybe over UDP? If I understand things correctly this data will be supplied in JSON format which can be easily consumed by either the Python data collection script and/or directly by the portal.
I have put together a proof-of-concept setup using a laptop and some really old Othernet hardware which I bought some time ago which in the past was in fact able to receive Inmarsat data using JAREO on Windows. I am having trouble getting reception from I-4 F3 as well as I-4 F4. It might be a bad LNA because swapping out the OEM unlabeled NooElec RTL2832U RTL-SDR with an old AirSpy Mini did not help things either. I think I will also try to locate a better satellite finding app and maybe change positions to see if that helps but I am thinking I need to start investing in some more up to date hardware in the very near future...

**Reply by jprochazka** [reply]
I will defiantly look into Airframes.io. It would be nice to offer a way to share such data with others. Not only L band data but over-the-air data from ACARSDEC as well which should be part of the scripts shortly after I get an antenna up.

@thebaldgeek appears to know what he is talking about for sure, and his site appears to be something I will be referring to as things progress. I actually bookmarked his [How to build an L-Band ground station](https://thebaldgeek.github.io/L-Band.html) page being I need something more durable than the three metal plate one the Othernet setup I have come with.

Forgot to add, gives me a reason to fire up a 3D printer again as well.
https://www.thingiverse.com/thing:4980180

Thanks for the input. Was great hearing from you!

**Comment by thebaldgeek** [comment]
Given that your project centers around ADSB (ie local) data, I would not go the L-Band Satdump way. Inmarsat data will not be in your area, L-Band does not have any aircraft positions and none of the messages you receive will relate to the aircraft in your area.

dumpVDL (or acarsdec) provides JSON data for aircraft in your area. Same range as ADSB.
You should be mindful to not just be duplicating acarsHUB. Just as you should _not_ duplicate adsb.im (I have had very bad experience with that software since its an all in one custom docker-like image. It takes away control of your system and makes any work in shell somewhat impossible and running any sort of ACARS along side of it is not an option - Your script install is much more flexible, powerful and maintainable).

**Reply by jprochazka** [reply]
For me, receiving L-Band is mostly out of curiosity. I fully agree that the data will not be of much interest for most users which is why whether or not it will become an option is up in the air. It comes down to if there is anyone else is interested in it other than me. As you said, when I was able to receive the data via JAREO back a few years there was not much data to be had which most user would be interested in. Keep in mind if it does make its way into the scripting as an option it will be up to the user to decide if the software is even installed.

Script support for Acarsdec is finished up and I have been successful at logging said data. There is an issue I need to address with the portal installation scripting which I noticed while adding a page to display ACARS data before I can merge the changes into the master branch. After this is wrapped up, I need to figure out if Acarsdec is capable of feeding data to multiple locations such as Acarsserv and Airframes.io at the same time. I would like to add the option to send data to Airframes.io if the user so desires as well as save data locally. It is also a good possibility that Acarsserv will be replaced in the future by the current Python script collecting ADS-B data in order to store this data within the existing database. 

Thanks for your comments regarding the flexibility of the project. As far as Docker and all-in-one images are concerned, do not expect to ever see this project evolve into something like that. This started out as a way to add graphs to existing FlightAware PiAware images back in the day and evolved from there. The goal has always been compatibility along with full control over what is installed. Docker and all-in-one images take this control away which for me is not acceptable.

**Comment by MCancellier** [comment]
One thing I would suggest for non-position data is to add dumpvdl2 along either ACARSDEC. I really believe both modes are important to support. I’m finding in my area (Colorado) that I receive more VDL2 traffic than ACARS. One airband antenna will get it all. On Aug 7, 2024, at 09:50, Joseph Prochazka ***@***.***> wrote:﻿
For me, receiving L-Band is mostly out of curiosity. I fully agree that the data will not be of much interest for most users which is why whether or not it will become an option is up in the air. It comes down to if there is anyone else is interested in it other than me. As you said, when I was able to receive the data via JAREO back a few years there was not much data to be had which most user would be interested in. Keep in mind if it does make its way into the scripting as an option it will be up to the user to decide if the software is even installed.
Script support for Acarsdec is finished up and I have been successful at logging said data. There is an issue I need to address with the portal installation scripting which I noticed while adding a page to display ACARS data before I can merge the changes into the master branch. After this is wrapped up, I need to figure out if Acarsdec is capable of feeding data to multiple locations such as Acarsserv and Airframes.io at the same time. I would like to add the option to send data to Airframes.io if the user so desires as well as save data locally. It is also a good possibility that Acarsserv will be replaced in the future by the current Python script collecting ADS-B data in order to store this data within the existing database.
Thanks for your comments regarding the flexibility of the project. As far as Docker and all-in-one images are concerned, do not expect to ever see this project evolve into something like that. This started out as a way to add graphs to existing FlightAware PiAware images back in the day and evolved from there. The goal has always been compatibility along with full control over what is installed. Docker and all-in-one images take this control away which for me is not acceptable.

**Comment by thebaldgeek** [comment]
I'm glad you mentioned dumpVDL. ACARS is hardly used in California as well, so I have shut down my ACARS and repurposed those SDRs to dumpVDL stations around the world.
Its also worth noting that dumpVDL does have aircraft Lat/Lon.
My site extracts them and plots them on a tar1090 map.
Use squawk code 15 filter and see what I am plotting: https://tbgmap.airframes.io/

I find L-Band of great interest. I ran a quick fun poll one time on X about which avgeeks like more, C or L Band the clear winner was L-Band.
So while it takes some parsing and filtering, there is a LOT of interesting traffic on L-band. It just takes some cross referencing.

**Comment by thebaldgeek** [comment]
Forgot to add that acarsdec can only support one feed. Its a real problem and there are a few work arounds which also are annoying.
dumpVDL supports as many as you want.

**Comment by jprochazka** [comment]
[DumpVLD2](https://github.com/szpajder/dumpvdl2) looks and sounds interesting and may very well be next up as far as decoder support goes. I have been lucky in that I have been getting a decent number of ACARS messages in northeast Ohio. Besides the obligatory position, software reports, and see you soon messages there have been a few miscellaneous messages such as a report of a clogged sink in a lavatory once. It would be interesting to see what VLD has to offer as well.

As for Acarsdec only servicing one endpoint I was afraid of this being the case and have looked into [UDPrepeater](https://github.com/UnionPacific/udp-repeater) as a possible solution to the problem. It seems like this would be the easiest solution only requiring Acarsdec data to be sent to the port on which UDPrepeater is listening then configuring UDPrepeater to forward said data along to its final destinations. I hate adding additional software but in order to feed multiple locations using Acarsdec it might be a necessary evil. Of course, this would be made available as an option if one wishes to feed a site such as Airframes.io while retaining data locally as well.

**Comment by jprochazka** [comment]
As a heads up, I started adding the option to install either dumpvdl2 or vdlm2dec. These changes can be tracked in the [vdlm2 branch](https://github.com/jprochazka/adsb-receiver/tree/vdlm2). I am reworking some of the logic used for device assignments and have not tested things thoroughly so I cannot guarantee the branch is in a working state at this time.

**Comment by jprochazka** [comment]
[DumpVLD2](https://github.com/szpajder/dumpvdl2) and [VDLM2DEC](https://github.com/TLeconte/vdlm2dec) support has been added.

https://github.com/jprochazka/adsb-receiver/discussions/608
